DDP-6297 allow prism admin to edit user activities

aglushko-broad opened the pull request:
If you're ok with the implemented approach I'll add unit tests for the admin editing panel
most of the removed code was added in terms of the PR https://github.com/broadinstitute/ddp-angular/pull/569

![image](https://user-images.githubusercontent.com/82094572/121302958-20f5da00-c903-11eb-8eb6-76431d8199f1.png)
![image](https://user-images.githubusercontent.com/82094572/121303671-0a9c4e00-c904-11eb-8a99-8c8a978a28b5.png)

![image](https://user-images.githubusercontent.com/82094572/121303889-49320880-c904-11eb-8896-532454bcee0d.png)
![image](https://user-images.githubusercontent.com/82094572/121303941-59e27e80-c904-11eb-9ba0-ae010df5ab91.png)

Commit:
DDP-6297 allow prism admin to edit user activities

aglushko-broad commented:
@MocanaAtBroad should we change the dashboard url approach from 
'dashboard/:userGuid' to 
'dashboard?user_guid=user_guid' 
It'd look more consistent

MocanaAtBroad commented:
Do we need to carry the user guid in the URL? We keep it in local store, right?

MocanaAtBroad commented:
When in Prism, when get redirected back to dashboard after hitting "Submit" at the end of the survey being edited, we don't show any activities. Perhaps we are losing the user guid being edited? (The user being edited is preserved in testboston).

aglushko-broad commented:
> Do we need to carry the user guid in the URL? We keep it in local store, right?

No, actually we don't (I put participation user data into session storage for better performance in the previous commit, but removed it later since it's sensitive data). See the details below in the comment

Commit:
DDP-6297 fix dashboard redirect at the end of user activities

aglushko-broad commented:
> When in Prism, when get redirected back to dashboard after hitting "Submit" at the end of the survey being edited, we don't show any activities. Perhaps we are losing the user guid being edited? (The user being edited is preserved in testboston).

oops, you're right. The problem was that actions redirects use query param and initial dashboard prism implementation used url param. I changed it to queryParam too for now.
testboston works differently: it adds participantGuid into the session (locates in local storage)
how it works:
testboston (old prism, only one user per search): admin searches by invitation id -> finds the user -> add participantGuid of found user guid to the localstorage (and add invitation code) -> reset the session guid when admin changes search string
angio and brain (new prism, multiple users per search): admin searches by any string -> finds the list of users -> navigates through the dashboard and actions taking participantGuid from url query param (and requests additional participant data by participantGuid  from the url)

actually, I'm starting to think that there is no advantage of having this new url param strategy. I remember we discussed it in the chat with Yufeng before the implementation and you mentioned that you expected it would look like that:  /prism/patient/PATIENT_GUID and I also thought it might be a good idea to have a participant guid in the url param. But now I think it's probably fine to have participantGuid in the session for the new prism as well as it was done for testboston prism. It allows simpler reuse and removing of the logic we implemented for using selectedUserGuid from url param.

@MocanaAtBroad  What do you think?

Commit:
DDP-6297 remove sensitive data from web storage

Commit:
DDP-6297 update css

Commit:
DDP-6297 fix indent

MocanaAtBroad commented:
> > When in Prism, when get redirected back to dashboard after hitting "Submit" at the end of the survey being edited, we don't show any activities. Perhaps we are losing the user guid being edited? (The user being edited is preserved in testboston).
> 
> oops, you're right. The problem was that actions redirects use query param and initial dashboard prism implementation used url param. I changed it to queryParam too for now.
> testboston works differently: it adds participantGuid into the session (locates in local storage)
> how it works:
> testboston (old prism, only one user per search): admin searches by invitation id -> finds the user -> add participantGuid of found user guid to the localstorage (and add invitation code) -> reset the session guid when admin changes search string
> angio and brain (new prism, multiple users per search): admin searches by any string -> finds the list of users -> navigates through the dashboard and actions taking participantGuid from url query param (and requests additional participant data by participantGuid from the url)
> 
> actually, I'm starting to think that there is no advantage of having this new url param strategy. I remember we discussed it in the chat with Yufeng before the implementation and you mentioned that you expected it would look like that: /prism/patient/PATIENT_GUID and I also thought it might be a good idea to have a participant guid in the url param. But now I think it's probably fine to have participantGuid in the session for the new prism as well as it was done for testboston prism. It allows simpler reuse and removing of the logic we implemented for using selectedUserGuid from url param.
> 
> @MocanaAtBroad What do you think?

When we GET activities, the URL used already includes the user guid, e.g.:
https://pepper-dev.datadonationplatform.org/pepper/v1/user/XXOWGMTY26BZ9IESDWI0*/studies/ANGIO/activities/VYFAC1HJN5

**XXOWGMTY26BZ9IESDWI0** is the user GUID, so redundant to also have it in a query param.

Perhaps it makes sense to have the GUID in URL going from the results to the dashboard? Maybe? If you think it makes sense go for it.

It looks like we store at least one result in storage after query, but we don't read it? (was looking at usage of SessionStorageService) . Could not see that we were reading it back but maybe I missed where. If we don't read it back we should get rid of it.

Otherwise, this is looking good and does everything that we need it to do as far as I can see.

Commit:
DDP-6297 keep selected user in the session instead of url param

aglushko-broad commented on ddp-workspace/projects/ddp-angio/src/app/app.module.ts:
```diff
@@ -96,65 +95,64 @@ config.prismRoute = 'prism';
+            }, () => {
+                resolve(null);
+            });
+        });
```
only indent changes

aglushko-broad commented on ddp-workspace/projects/ddp-angio/src/app/app.module.ts:
```diff
@@ -96,65 +95,64 @@ config.prismRoute = 'prism';
-    MoreDetailsComponent,
-    PrismComponent,
-    WelcomeComponent,
-    DashboardComponent
```
DashboardComponent was removed, rest - indent

aglushko-broad commented on ddp-workspace/projects/ddp-brain/src/app/app.module.ts:
```diff
@@ -96,73 +95,72 @@ sdkConfig.lookupPageUrl = AppRoutes.Prism;
+            }, () => {
+                resolve(null);
+            });
+        });
```
only indent

aglushko-broad commented on ddp-workspace/projects/ddp-brain/src/app/app.module.ts:
```diff
@@ -96,73 +95,72 @@ sdkConfig.lookupPageUrl = AppRoutes.Prism;
-    ModalImageComponent,
-    LearnMoreSectionComponent,
-    PrismComponent,
-    DashboardComponent,
```
DashboardComponent was removed, rest - indent

aglushko-broad commented on ddp-workspace/projects/ddp-sdk/src/lib/components/dashboard.component.ts:
```diff
@@ -1,36 +1,24 @@
-    </p>
-    <ddp-user-activities [studyGuid]="studyGuid"
-                         [selectedUserGuid]="selectedUserGuid"
-                         [displayedColumns]="activitiesColumns"
```
selectedUserGuid and displayedColumns were removed here only

aglushko-broad commented on ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts:
```diff
@@ -14,140 +15,145 @@ import { SessionStorageService } from '../../services/sessionStorage.service';
-  public isSearchLoading = false;
-  public isSearchDebounce = false;
-  public MIN_SEARCH_LENGTH = 4;
-  public dashboardRoute: string;
```
dashboardRoute was removed

aglushko-broad commented on ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts:
```diff
@@ -14,140 +15,145 @@ import { SessionStorageService } from '../../services/sessionStorage.service';
+        private participantsSearch: ParticipantsSearchServiceAgent,
+        private router: Router,
+        private storageService: SessionStorageService,
+        private sessionService: SessionMementoService,
```
SessionMementoService injected

aglushko-broad commented on ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts:
```diff
@@ -14,140 +15,145 @@ import { SessionStorageService } from '../../services/sessionStorage.service';
+        @Inject('ddp.config') private config: ConfigurationService,
     ) {
-    this.displayedColumns = this.config.prismColumns;
-    this.dashboardRoute = this.config.prismDashboardRoute;
```
dashboardRoute was removed

aglushko-broad commented on ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts:
```diff
@@ -14,140 +15,145 @@ import { SessionStorageService } from '../../services/sessionStorage.service';
+        return this.isSearchLoading || this.dataSource.data.length <= this.paginationSizes[0];
+    }
+
+    public clickOnUser(user: SearchParticipant): void {
```
clickOnUser method was added

MocanaAtBroad reviewed (dismissed):
LGTM!

Commit:
Merge branch 'develop' into DDP-6297-prism-edit-user-activities

# Conflicts:
#	ddp-workspace/projects/ddp-angio/src/assets/i18n/en.json
#	ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts
